### PR TITLE
Adding a permission per partner to be able to configure "copy entitelmen...

### PR DESCRIPTION
**Pull request by noam-arad**
...t" for kaltura live streams and if the flag is on I make sure the categories (and entitlements) are copied (synced).

PLAT-2744

### CODE REVIEW

File: alpha/lib/data/live/kLiveEntryRecordingOptions.php [hunk 1]
```diff
@@ -23,10 +23,15 @@ public function setShouldCopyEntitlement($shouldCopyEntitlement)
 	}
 	
 	/**
+	 * @var int
 	 * @return boolean
 	 */
-	public function getShouldCopyEntitlement()
+	public function getShouldCopyEntitlement($partnerId)
 	{
+		if ( PermissionPeer::isValidForPartner(PermissionName::FEATURE_LIVE_STREAM_COPY_ENTITELMENTS, $partnerId) )
```
Comment (tan-tan-kanarek) [inline alpha/lib/data/live/kLiveEntryRecordingOptions.php]: This way you prevent disabling it using the API.
I suggest that you will ovveride KalturaEntryService::prepareEntryForInsert in KalturaLiveEntryService and check if this attribute is null, and only in such case, use the partner defaults.

Comment (noam-arad) [inline alpha/lib/data/live/kLiveEntryRecordingOptions.php]: @tan-tan-kanarek Please review changes to fit your comments

File: api_v3/lib/types/entry/KalturaLiveEntry.php [hunk 2]
```diff
@@ -126,7 +126,21 @@ public function toInsertableObject($sourceObject = null, $propsToSkip = array())
 	{
 		if(is_null($this->recordStatus))
 			$this->recordStatus = (PermissionPeer::isValidForPartner(PermissionName::FEATURE_LIVE_STREAM_RECORD, kCurrentContext::getCurrentPartnerId()) ? KalturaRecordStatus::APPENDED : KalturaRecordStatus::DISABLED);
-			
+
+		if (is_null($this->recordingOptions))
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: There is no reason to instantiate recordingOptions if FEATURE_LIVE_STREAM_COPY_ENTITELMENTS permission is off.

File: api_v3/lib/types/entry/KalturaLiveEntry.php [hunk 3]
```diff
@@ -126,7 +126,21 @@ public function toInsertableObject($sourceObject = null, $propsToSkip = array())
...
+		{
+			$this->recordingOptions = new kLiveEntryRecordingOptions();
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: You should instantiate KalturaLiveEntryRecordingOptions and not kLiveEntryRecordingOptions.

**Pull request merged**